java-checks-test-sources/default/src/main/files/non-compiling/checks/SunPackagesUsedCheckSample.java

```diff
@@ -16,4 +18,18 @@ public Object uselessMethod() {
     }
     return null;
   }
+
+  // SONARJAVA-4698: False positive when variable is named "sun"
+  public void fooWithFieldNamedSun() {
+    sun.toString(); // Compliant - FP: "sun" is a field of type Object, not a sun.* package class
+  }
+
+  public void barWithParameterNamedSun(Object sun) {
+    sun.toString(); // Compliant - FP: "sun" is a parameter of type Object, not a sun.* package class
```

Comment:
I don't think we should mention "FP" in the comment, it sounds confusing to me: I would rather leave only `// Compliant - "sun" is a field of type Object, not a sun.* package class`